lib/assign-properties.js

```diff
@@ -56,16 +56,36 @@ if (sourceAvailable) {
   }
 }
 
+function giveAccessorSuper(descriptor, name, target) {
+  let old = descriptor[name];
+  let superMethod = target[name];
+
+  descriptor[name] = function() {
+    this._super = superMethod;
+    return old.apply(this, arguments);
+  }
+}
+
 module.exports = function assignProperties(target, options) {
   let value;
 
-  for (let key in options) {
-    value = options[key];
+  Object.getOwnPropertyNames(options).forEach(key => {
+    let descriptor = Object.getOwnPropertyDescriptor(options, key);
+    let value = descriptor.value;
 
     if (typeof value === 'function' && hasSuper(value)) {
-      target[key] = giveMethodSuper(getPrototypeOf(target), key, value);
-    } else {
-      target[key] = value;
+      descriptor.value = giveMethodSuper(getPrototypeOf(target), key, value);
+    } else if (value === undefined){
+      if (typeof descriptor.get === 'function' && hasSuper(descriptor.get)) {
+        giveAccessorSuper(descriptor, 'get', Object.getOwnPropertyDescriptor(target, key));
+      }
+
+      if (typeof descriptor.set === 'function' && hasSuper(descriptor.set)) {
+        giveAccessorSuper(descriptor, 'set', Object.getOwnPropertyDescriptor(target, key));
```

Comment:
Can you add a test for multiple level inheritance? Seems like this will only work if `target` itself has own property with the same name, but if "target" was `Bar` in the below:

```
class Foo extends CoreObject {
  get thing() { return 'foo stuff!'; }
}

class Bar extends Foo {}

const Derp = Bar.extend({
  get thing() { return this._super(); }
});
```